x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/__test__/waterfall_helpers.test.ts

```diff
@@ -148,6 +148,98 @@ describe('waterfall_helpers', () => {
     });
   });
 
+  describe('getWaterfall - legacy', () => {
+    const hits = [
+      {
+        processor: { event: 'transaction' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-node' },
+        transaction: {
+          duration: { us: 49660 },
+          name: 'GET /api',
+          id: 'myTransactionId1'
+        },
+        timestamp: { us: 1549324795784006 }
+      } as Transaction,
+      {
+        processor: { event: 'span' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-ruby' },
+        transaction: { id: 'myTransactionId2' },
+        timestamp: { us: 1549324795825633 },
+        span: {
+          duration: { us: 481 },
+          name: 'SELECT FROM products',
+          id: 'mySpanIdB'
+        }
+      } as Span,
+      {
+        processor: { event: 'span' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-ruby' },
+        transaction: { id: 'myTransactionId2' },
+        span: {
+          duration: { us: 6161 },
+          name: 'Api::ProductsController#index',
+          id: 'mySpanIdA'
+        },
+        timestamp: { us: 1549324795824504 }
+      } as Span,
+      {
+        processor: { event: 'span' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-ruby' },
+        transaction: { id: 'myTransactionId2' },
+        span: {
+          duration: { us: 532 },
+          name: 'SELECT FROM product',
+          id: 'mySpanIdC'
+        },
+        timestamp: { us: 1549324795827905 }
+      } as Span,
+      {
+        processor: { event: 'span' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-node' },
+        transaction: { id: 'myTransactionId1' },
+        span: {
+          duration: { us: 47557 },
+          name: 'GET opbeans-ruby:3000/api/products',
+          id: 'mySpanIdD'
+        },
+        timestamp: { us: 1549324795785760 }
+      } as Span,
+      {
+        processor: { event: 'transaction' },
+        trace: { id: 'myTraceId' },
+        service: { name: 'opbeans-ruby' },
+        transaction: {
+          duration: { us: 8634 },
+          name: 'Api::ProductsController#index',
+          id: 'myTransactionId2'
+        },
+        timestamp: { us: 1549324795823304 }
+      } as Transaction
+    ];
+    it('should return waterfall', () => {
```

Comment:
Can we be more specific about how this behaviour is different? Eg. "should return waterfall sorted by start time" (if that's the behaviour)